### PR TITLE
Added docs + docstrings

### PR DESCRIPTION
**Pull request by kushalbakshi**

### CODE REVIEW

**Comment by kushalbakshi** [comment]
All the pages are populated with content in what should be its final version. Will be tagging up with Drew tomorrow to try to render it locally and proof-read/check links

File: docs/.docker/pip_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+mkdocs-material
+mkdocs-redirects
+mkdocstrings
+mkdocstrings-python
+mike
+mdx-truly-sane-lists
+mkdocs-gen-files
+mkdocs-literate-nav
+mkdocs-exclude-search
```
Comment (CBroz1) [inline docs/.docker/pip_requirements.txt:9]: Please add [automated versioning to the citation page](https://github.com/datajoint/element-deeplabcut/commit/352c44f98325cb79a249026d84310bff8417e7e2)

File: docs/src/concepts.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Concepts
+
+## Modeling the Brain
+
+Studies of brain anatomy and cellular morphology were once confined to single-subject studies wherein in vitro slices were compared to develop a complete model of the brain. With dyes, researchers could compare recordings to the reconstructed location of a given recording device. Through clever experimental design, researchers could further associate highly localized regions to their task-dependent function. Advances in 3D modeling have permitted parallel advances in multi-subject averaged anatomical models. An atlas serves as a shared reference frame for a given species. [The Allen Institute](https://mouse.brain-map.org/) has been a leader in the development of mouse atlases since 2007<sup>[1](#references)</sup>, with regular updates<sup>[2](#references)</sup> that provide researchers with a shared reference frame in the study of functional neuroanatomy.
```
Comment (CBroz1) [inline docs/src/concepts.md:5]: 1. Please hard wrap to make future edits easier to review
2. How do the footnotes look in local deployment? No Issues? I know mkdocs has their own extension for [footnotes](https://squidfunk.github.io/mkdocs-material/reference/footnotes/#on-multiple-lines), but I wouldn't add it if this gets the job done

File: docs/src/concepts.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
++ `coordinate_framework` ingests standard atlases and retains voxel-based lookup tables, including references to brain region information, acronyms and standardized color codes.
++ `electrode_localization` pairs the above reference tables with Neuropixels probe location data in a `probe` schema, optionally from [Element Array Ephys](https://github.com/datajoint/element-array-ephys)
+
+## Key Partnerships
+
+In coordination with both the Mesoscale Activity Project and the International Brain Lab, DataJoint developed mechanisms for pairing recording coordinates with the published atlases in project-specific cases that have since been generalized.
+
+## Pipeline Development and Limitations
+
+The Element's table architecture was generalized from existing project-specific pipelines to load standardized Kilosort `channel_locations.json` files as well as the Allen Institute's [atlas files](https://community.brain-map.org/t/allen-mouse-ccf-accessing-and-using-related-data-and-tools/359), and pair these data sets.
```
Comment (CBroz1) [inline docs/src/concepts.md:24]: @sidhulyalkar - Are `channel_location` jsons automatically generated with Kilosort? In my reading for the diagrams, it looked like that was an IBL app thing.

File: docs/src/concepts.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+One notable consession was made in development: acronyms in DataJoint do not perfectly map on to the Allen Institute's published standard. By default, DataJoint databases are not case sensitive. Instead, acronyms are convered to [snake case](https://en.wikipedia.org/wiki/Snake_case) to avoid naming collisions. While we depart from the standard, preliminary interviews with users indicate no bias toward the official standard. Visit our [localization notebook](https://github.com/datajoint/workflow-array-ephys/blob/main/notebooks/08-electrode-localization.ipynb) for a demonstration of converting between the case sensitive and snake case standards. 
+
+## Roadmap
+
+Further development of this Element is community driven. Upon user requests we will continue adding features to this Element.
```
Comment (CBroz1) [inline docs/src/concepts.md:30]: @sidhulyalkar - Can you think of something we can add here?
```suggestion
Further development of this Element is community driven. Upon user requests we will continue adding features to this Element, such as improved topological referencing.
```
What comes to mind for me is better linking across child and parent regions, or importing only a subset of the full CCF

File: element_electrode_localization/coordinate_framework.py [hunk 5]
```diff
@@ -52,11 +68,26 @@ class Voxel(dj.Part):
...
+    """
     definition = """
     -> CCF
     """
 
     class BrainRegion(dj.Part):
+        """Brain region information.
+        
+        Attributes:
+            master (foreign key): BrainRegionAnnotion primary key.
```
Comment (CBroz1) [inline element_electrode_localization/coordinate_framework.py:84]: I don't think there's a right way to do this (and I'd also like to automate this down the road, so it'll get rewritten here), but the convention I developed was to write the table name instead of master [e.g.](https://github.com/datajoint/element-deeplabcut/blob/main/element_deeplabcut/model.py#L121) - It looks like you've done the same elsewhere

File: element_electrode_localization/coordinate_framework.py [hunk 6]
```diff
@@ -105,24 +154,14 @@ class ParentBrainRegion(dj.Lookup):
...
-        ontology_csv_filepath='./data/query.csv')
-
-    For an example Allen brain atlas for mouse, see:
-    http://download.alleninstitute.org/informatics-archive/current-release/mouse_ccf/annotation/ccf_2017
-
-    For the structure/ontology tree, see:
-    https://community.brain-map.org/t/allen-mouse-ccf-accessing-and-using-related-data-and-tools/359
-    (particularly the ontology file downloadable as CSV)
+    """Load CCF annotation
+
```
Comment (CBroz1) [inline element_electrode_localization/coordinate_framework.py:158]: Google style allows a more substantive description after the oneliner, and then an Example section. Please add back these pieces.

File: element_electrode_localization/electrode_localization.py [hunk 7]
```diff
@@ -24,24 +24,14 @@ def activate(
...
-        :param linking_module: a module name or a module containing the
-         required dependencies to activate the `electrode_localization` element:
-            Upstream tables:
-                + ProbeInsertion: table referenced by ElectrodePosition, typically
-                                  identifying a Probe Insertion instance
-                + probe: the probe schema - from element-array-ephys
+    """Activates the `electrode_localization` and `coordinate_framework` schemas. 
+
+    Args:
+        electrode_localization_schema_name (str): A string containing the name of the electrode_localization schema.
```
Comment (CBroz1) [inline element_electrode_localization/electrode_localization.py:30]: please wrap to 88 where possible. For long arg descriptions, we can just increase the indent on the next line, not wrap to the `:`
```suggestion
        electrode_localization_schema_name (str): A string containing the name of the 
            electrode_localization schema.
```

**Review by CBroz1 — CHANGES_REQUESTED** [review]

File: docs/.docker/pip_requirements.txt (hunk 1 above)
Comment (kushalbakshi) [reply docs/.docker/pip_requirements.txt:9]: Done

File: docs/src/concepts.md (hunk 2 above)
Comment (kushalbakshi) [reply docs/src/concepts.md:5]: There are no issues in local deployment with that style for footnotes

File: element_electrode_localization/coordinate_framework.py (hunk 5 above)
Comment (kushalbakshi) [reply element_electrode_localization/coordinate_framework.py]: This has been updated to contain the name of the master table

File: element_electrode_localization/coordinate_framework.py (hunk 6 above)
Comment (kushalbakshi) [reply element_electrode_localization/coordinate_framework.py:166]: Added

File: element_electrode_localization/electrode_localization.py (hunk 7 above)
Comment (kushalbakshi) [reply element_electrode_localization/electrode_localization.py:30]: Should be updated (but I'm still not sure if its wrapping correctly)

Comment (CBroz1) [reply element_electrode_localization/electrode_localization.py:30]: It doesn't seem to be updated, but you can do so by committing the suggestion above

File: docs/src/concepts.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+The Element's table architecture was generalized from existing project-specific pipelines to load standardized Kilosort `channel_locations.json` files as well as the Allen Institute's [atlas files](https://community.brain-map.org/t/allen-mouse-ccf-accessing-and-using-related-data-and-tools/359), and pair these data sets.
+
+One notable consession was made in development: acronyms in DataJoint do not perfectly map on to the Allen Institute's published standard. By default, DataJoint databases are not case sensitive. Instead, acronyms are convered to [snake case](https://en.wikipedia.org/wiki/Snake_case) to avoid naming collisions. While we depart from the standard, preliminary interviews with users indicate no bias toward the official standard. Visit our [localization notebook](https://github.com/datajoint/workflow-array-ephys/blob/main/notebooks/08-electrode-localization.ipynb) for a demonstration of converting between the case sensitive and snake case standards. 
+
+## Roadmap
+
+Further development of this Element is community driven. Upon user requests we will continue adding features to this Element, such as improved topological referencing.
```
Comment (CBroz1) [inline docs/src/concepts.md:30]: @sidhulyalkar - please confirm the language I suggested here. What I mean is a feature that would allow researchers to better see the relationships between regions and subregions in the ccf hierarchy

File: element_electrode_localization/coordinate_framework.py [hunk 9]
```diff
@@ -67,6 +99,12 @@ class BrainRegion(dj.Part):
         """
 
     class Voxel(dj.Part):
+        """Voxel information from CCF.
+
+        Attributes:
+            master.BrainRegion (foreign key): BrainRegionAnnotation.BrainRegion primary key.
```
Comment (CBroz1) [inline element_electrode_localization/coordinate_framework.py:105]:
```suggestion
            BrainRegion (foreign key): BrainRegionAnnotation.BrainRegion primary key.
```

File: element_electrode_localization/coordinate_framework.py [hunk 10]
```diff
@@ -92,6 +136,13 @@ def voxel_query(self, x=None, y=None, z=None):
 
 @schema
 class ParentBrainRegion(dj.Lookup):
+    """Hierarchical structure between the brain regions.
+
+    Attributes:
+        BrainRegionAnnotation.BrainRegion (foreign key): BrainRegionAnnotation.BrainRegion primary key.
+        BrainRegionAnnotation.BrainRegion.proj(parent='acronym') (query): fetches brain region acronym
```
Comment (CBroz1) [inline element_electrode_localization/coordinate_framework.py:143]: in practice, proj looks like a new field renamed
```suggestion
        Parent (query): parent brain region acronym from BrainRegion table
```

**Review by CBroz1 — CHANGES_REQUESTED** [review]
Looks good. Few minor changes and a couple items I'd like @sidhulyalkar to provide input on

File: docs/src/concepts.md (hunk 3 above)
Comment (sidhulyalkar) [reply docs/src/concepts.md:24]: I don't believe the 'channel_locations.json' is generated with kilosort, it seems to be specifically generated by the IBL app 👍🏻

File: docs/src/concepts.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+## Key Partnerships
+
+In coordination with both the Mesoscale Activity Project and the International Brain Lab, DataJoint developed mechanisms for pairing recording coordinates with the published atlases in project-specific cases that have since been generalized.
+
+## Pipeline Development and Limitations
+
+The Element's table architecture was generalized from existing project-specific
+pipelines such as International Brain Laboratory's `channel_locations.json` as well as the Allen
+Institute's [atlas
```
Comment (kushalbakshi) [inline docs/src/concepts.md:26]: Is this a better way to word this section, @sidhulyalkar?

File: docs/src/concepts.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+files](https://community.brain-map.org/t/allen-mouse-ccf-accessing-and-using-related-data-and-tools/359),
```
Comment (CBroz1) [inline docs/src/concepts.md:27]: careful of linebreaks in links

Comment (kushalbakshi) [reply docs/src/concepts.md:27]: It works in my local deployment

Comment (CBroz1) [reply docs/src/concepts.md:27]: my recollection is that this breaks when deployed on gh-pages, but that was me working on my personal site, without the dockerization. We can make a note to confirm after merged or I can pull to my fork to launch gh-pages

File: docs/src/concepts.md (hunk 8 above)
Comment (sidhulyalkar) [reply docs/src/concepts.md:34]: Yes that sounds appropriate. Could reword it to say "improved regional and subregional based topological referencing" for clarity.

File: docs/src/concepts.md (hunk 12 above)
Comment (kushalbakshi) [reply docs/src/concepts.md:27]: I can confirm after the merge if that's easier

File: docs/src/concepts.md (hunk 11 above)
Comment (sidhulyalkar) [reply docs/src/concepts.md:26]: Maybe something like this: The Element's table architecture was generalized from existing project-specific
pipelines such as International Brain Laboratory's [iblapps](https://github.com/int-brain-lab/iblapps/wiki/) as well as the Allen
Institute's [atlas files](https://community.brain-map.org/t/allen-mouse-ccf-accessing-and-using-related-data-and-tools/359)

Specifically, the element depends on a file generated using the IBLapps GUI, 'channel_locations.json'.

Comment (kushalbakshi) [reply docs/src/concepts.md:26]: Thank you!

**Review by CBroz1 — APPROVED** [review]
Thanks for all your work here @kushalbakshi ! Please tag once merged

File: docs/src/concepts.md (hunk 12 above)
Comment (kushalbakshi) [reply docs/src/concepts.md]: This works on the live site
